docs/resources/endpoint_security_group.md

```diff
@@ -7,100 +10,247 @@ description: |-
 ---
 
-# aci_endpoint_security_group
+# aci_endpoint_security_group #
 
 Manages ACI Endpoint Security Group
 
-## API Information
 
-- `Class` - fvESg
-- `Distinguished Name` - uni/tn-{name}/ap-{name}/esg-{name}
 
-## GUI Information
+## API Information ##
 
-- `Location` - Tenants > {tenant_name} > Application Profiles > Endpoint Security Groups
+* Class: [fvESg](https://pubhub.devnetcloud.com/media/model-doc-latest/docs/app/index.html#/objects/fvESg/overview)
 
-## Example Usage
+* Supported in ACI versions: 5.0(1k) and later.
+
+* Distinguished Name Format: `uni/tn-{name}/ap-{name}/esg-{name}`
+
+## GUI Information ##
+
+* Location: `Tenants -> Application Profiles -> Endpoint Security Groups`
+
+## Example Usage ##
+
+The configuration snippet below creates a Endpoint Security Group with only required attributes.
+
+```hcl
+
+resource "aci_endpoint_security_group" "example_application_profile" {
+  parent_dn = aci_application_profile.example.id
+  name      = "test_name"
+}
+
+```
+The configuration snippet below shows all possible attributes of the Endpoint Security Group.
+
+!> This example might not be valid configuration and is only used to show all possible attributes.
 
 ```hcl
-resource "aci_endpoint_security_group" "example" {
-  application_profile_dn  = aci_application_profile.example.id
-  name  = "example"
-  description = "from terraform"
-  annotation = "orchestrator:terraform"
-  name_alias = "example"
-  match_t = "AtleastOne"
-  pc_enf_pref = "unenforced"
-  pref_gr_memb = "exclude"
-
-  relation_fv_rs_scope = aci_vrf.example.id
-
-  relation_fv_rs_cons {
-    prio = "unspecified"
-    target_dn = aci_contract.example.id
-  }
-
-  relation_fv_rs_cons_if {
-    prio = "unspecified"
-    target_dn = aci_imported_contract.example.id
-  }
-
-  relation_fv_rs_cust_qos_pol = aci_resource.example.id
-
-  relation_fv_rs_intra_epg = [aci_contract.example.id]
-
-  relation_fv_rs_prov {
-    match_t = "AtleastOne"
-    prio = "unspecified"
-    target_dn = aci_contract.example.id
-  }
-
-  relation_fv_rs_sec_inherited = [aci_application_epg.example.id]
+
+resource "aci_endpoint_security_group" "full_example_application_profile" {
+  parent_dn              = aci_application_profile.example.id
+  annotation             = "annotation"
+  description            = "description"
+  exception_tag          = "exception_tag"
+  match_criteria         = "All"
+  name                   = "test_name"
+  name_alias             = "name_alias"
+  intra_esg_isolation    = "enforced"
+  preferred_group_member = "exclude"
+  admin_state            = "no"
+  relation_to_consumed_contracts = [
+    {
+      annotation    = "annotation_1"
+      priority      = "level1"
+      contract_name = aci_contract.example.name
+    }
+  ]
+  relation_to_consumed_contract_interfaces = [
+    {
+      annotation              = "annotation_1"
+      priority                = "level1"
+      contract_interface_name = aci_contract_interface.example.name
```

Comment:
I don't think we have a resource aci_contract_interface? Is this the right value?